tests/integration_tests/assets/__init__.py

```diff
@@ -0,0 +1,16 @@
+import os
+from collections import namedtuple
+from pathlib import Path
+
+ASSET_DIR = Path(os.path.dirname(os.path.realpath(__file__)))
+PRIVATE_KEY_PATH = ASSET_DIR / 'test_id_rsa'
+PUBLIC_KEY_PATH = ASSET_DIR / 'test_id_rsa.pub'
+
+
+def get_test_keypair():
```

Comment:
Do we want to make this interface `get_asset(asset_id: str) -> Any` to discourage ourselves from littering this with inconsistent utility functions in future? (The implementation doesn't need to be generic; `if asset_id != "test_keypair": raise Exception("Unknown asset")` seems like a fine placeholder for us to fix when we need to.)

(Perhaps this would be better as a fixture, to enforce a consistent single "entrypoint"; that wouldn't work at module-level for USER_DATA mark content, however, and I think that's a puzzle to be solved another time.)

Reply:
Hmmmm, I think I'd rather have 20 random utility functions than 20 random strings in which we check which asset is tied to which string. Even if we had something like an Enum, that's essentially just adding a level of indirection to get to the private function doing what could be done with a public function. Is there a reason you want to restrict the number of entry points?

Reply:
> Hmmmm, I think I'd rather have 20 random utility functions than 20 random strings in which we check which asset is tied to which string.

I don't think these are equivalent: utility functions can have completely different signatures to one another, whereas strings are flat identifiers. I think that's the core of my concern: if these can take parameters then they can behave in substantially different ways to one another.

> Even if we had something like an Enum, that's essentially just adding a level of indirection to get to the private function doing what could be done with a public function. Is there a reason you want to restrict the number of entry points?

Experience tells me that we'll end up with test logic being implemented in functions in this file, which means (a) our tests are harder to read/write because they're more spread out, and (b) the functions won't be (easily) usable outside of the specific test they were written for (because they'll be written with whatever that test author sees as the "assets interface" at that moment in time).

That said, the `-> Any` in my proposal isn't ideal; it makes it impossible to reason about what `get_asset` would return without going and reading the assets code.

You aren't introducing an issue here, so we don't need to take care of this to get this in; let's be mindful of such issues as we introduce more assets in future but for now: YAGNI (WAGNI?).